### PR TITLE
Add missing Vulkan cache flushes and invalidates

### PR DESCRIPTION
**Pull request by gary-sweet**
## Description

Without these flushes and invalidates, I see missing data on readback and bad text in the overlay.
Fixes issue #1090

### CODE REVIEW

**Review by baldurk — APPROVED** [review]
Thanks, this looks good!

Are there no flushes/invalidates needed for the initial states? That's the other common place where memory is mapped for upload and readback.

**Comment by gary-sweet** [comment]
> Are there no flushes/invalidates needed for the initial states?

It's entirely possible that I may have missed some. It's the first time I've looked at this codebase.

**Comment by baldurk** [comment]
That's entirely fair, I wasn't sure how you'd identified the maps that needed flushes so I figured I'd flag it up just in case. If you look in `vk_initialstate.cpp` there are a couple of map calls used to readback the contents of memory and textures at frame start, and then upload them again during replay. If you've captured and replayed any non-trivial program you should have run through this code since all memory that isn't uploaded mid-frame gets saved out here.

I can try and make a fix myself, but I'd need you to verify it either way. The changes are fine as-is though, so unless you want me to hold off I'll merge when the CI is green.

**Comment by gary-sweet** [comment]
I'm happy to take a look and add a new PR to cover that case.

**Pull request merged**
